packages/video_player/video_player_avfoundation/darwin/Classes/FVPVideoPlayerPlugin.m

```diff
@@ -20,66 +21,72 @@ @interface FVPFrameUpdater : NSObject
 @property(nonatomic, weak, readonly) NSObject<FlutterTextureRegistry> *registry;
 // The output that this updater is managing.
 @property(nonatomic, weak) AVPlayerItemVideoOutput *videoOutput;
-#if TARGET_OS_IOS
-- (void)onDisplayLink:(CADisplayLink *)link;
-#endif
+// The last time that has been validated as avaliable according to hasNewPixelBufferForItemTime:.
+@property(nonatomic, assign) CMTime lastKnownAvailableTime;
+// If YES, the engine is informed that a new texture is available any time the display link
+// callback is fired, regardless of the videoOutput state.
+//
+// TODO(stuartmorgan): Investigate removing this; it exists only to preserve existing iOS behavior
+// while implementing macOS, but iOS should very likely be doing the check as well.
+@property(nonatomic, assign) BOOL skipBufferAvailabilityCheck;
 @end
 
 @implementation FVPFrameUpdater
 - (FVPFrameUpdater *)initWithRegistry:(NSObject<FlutterTextureRegistry> *)registry {
   NSAssert(self, @"super init cannot be nil");
   if (self == nil) return nil;
   _registry = registry;
+  _lastKnownAvailableTime = kCMTimeInvalid;
   return self;
 }
 
-#if TARGET_OS_IOS
-- (void)onDisplayLink:(CADisplayLink *)link {
-  // TODO(stuartmorgan): Investigate switching this to displayLinkFired; iOS may also benefit from
-  // the availability check there.
-  [_registry textureFrameAvailable:_textureId];
-}
-#endif
-
 - (void)displayLinkFired {
-  // Only report a new frame if one is actually available.
-  CMTime outputItemTime = [self.videoOutput itemTimeForHostTime:CACurrentMediaTime()];
-  if ([self.videoOutput hasNewPixelBufferForItemTime:outputItemTime]) {
+  // Only report a new frame if one is actually available, or the check is being skipped.
+  BOOL reportFrame = NO;
+  if (self.skipBufferAvailabilityCheck) {
+    reportFrame = YES;
+  } else {
+    CMTime outputItemTime = [self.videoOutput itemTimeForHostTime:CACurrentMediaTime()];
+    if ([self.videoOutput hasNewPixelBufferForItemTime:outputItemTime]) {
+      _lastKnownAvailableTime = outputItemTime;
+      reportFrame = YES;
+    }
```

Comment:
Reset in else? 

```
else {
  _lastKnownAvailableTime = kCMTimeInvalid;
  reportFrame = NO;
}
```

Reply:
`reportFrame` is a local initialized to `NO`, so there's no need to set it to `NO`.

Why would we reset `_lastKnownAvailableTime`? If we don't have a new time, then the last known time is still the last known time.